lib/buffer.js

```diff
@@ -64,7 +64,17 @@ function alignPool() {
  * much breakage at this time. It's not likely that the Buffer constructors
  * would ever actually be removed.
  **/
+var newBufferWarned = false;
 function Buffer(arg, encodingOrOffset, length) {
+  if (!new.target && !newBufferWarned) {
+    newBufferWarned = true;
+    process.emitWarning(
+      'Using Buffer without `new` will soon stop working. ' +
+      'Use `new Buffer`, or preferably ' +
+      'Buffer.from, Buffer.allocUnsafe or Buffer.alloc instead.',
```

Comment:
I imagine users who don't keep track of Node.js development would be thoroughly confused by a message that says they shouldn't call `Buffer` without `new` and then tells them to use something entirely different. They would wonder whether they can just use `new Buffer` and if not, why they don't get such a message with `new Buffer`.